src/mesh/voxel_cloud.rs

```diff
@@ -228,6 +232,124 @@ impl VoxelCloud {
         }
     }
 
+    /// Computes boolean intersection (logical AND operation) of the current and
+    /// another Voxel cloud. The current Voxel cloud will be mutated and resized
+    /// to the size and position of an intersection of the two Voxel clouds'
+    /// volumes.
+    pub fn boolean_intersection(&mut self, other: &VoxelCloud) {
+        // Find volume common to both voxel clouds.
+        if let Some(self_volume_bounding_box) = self.volume_bounding_box() {
+            if let Some(other_volume_bounding_box) = other.volume_bounding_box() {
+                if let Some(bounding_box) = BoundingBox::intersection(
+                    [self_volume_bounding_box, other_volume_bounding_box].iter(),
+                ) {
+                    // Resize (keep or shrink) the existing voxel cloud so that that can
+                    // possibly contain intersection voxels.
+                    self.resize_to_voxel_space_bounding_box(&bounding_box);
+
+                    let block_start = bounding_box.minimum_point();
+                    let block_dimensions = bounding_box.diagonal();
+                    // Iterate through the block of space common to both voxel clouds.
+                    for i in 0..self.voxel_map.len() {
+                        let cartesian_coords = absolute_voxel_to_cartesian_coords(
+                            &one_dimensional_to_absolute_three_dimensional_coordinate(
+                                i,
+                                &block_start,
+                                &block_dimensions,
+                            )
+                            .expect("The current voxel map out of bounds"),
+                            self.voxel_dimensions,
+                        );
+
+                        // Perform boolean AND on voxel states of both voxel clouds.
+                        self.voxel_map[i] &= other
+                            .voxel_at_cartesian_coords(&cartesian_coords)
+                            .unwrap_or(false);
```

Comment:
I'm worried we can accidentally hit an out-of-bounds voxel due to possible different voxel dimensions. not sure, just a gut feeling it can happen and this autofix here seems good enough

Reply:
You are right. I forgot that the coords were cartesian. 👍